api/v1alpha1/datadogagent_types.go

```diff
@@ -760,7 +760,16 @@ type ClusterAgentConfig struct {
 	// Autodiscovery via Kube Service annotations is automatically enabled
 	ClusterChecksEnabled *bool `json:"clusterChecksEnabled,omitempty"`
 
-	// Enables this to start event collection from the Kubernetes API
+	// Enable this to start the Kubernetes State Metrics Core check.
+	// TODO add reference to the documentation.
+	// +optional
+	KubeStateMetricsCoreEnabled *bool `json:"kubeStateMetricsCoreEnabled,omitempty"`
+
+	// To override the configuration for the default Kubernetes State Metrics Core check.
+	// Must point to a ConfigMap containing a valid cluster check configuration.
+	KubeStateMetricsCoreConf *string `json:"kubeStateMetricsCoreConf,omitempty"`
+
+	// Enable this to start event collection from the kubernetes API
```

Comment:
Can we create a struct to agreggate the `KubeStateMetricsCore` Integration

```golang
type KubeStateMetricsCore struct {
   Enabled *bool
   Conf *conf
}
```
It will be more future proof. 

Since https://github.com/DataDog/datadog-operator/pull/178 was merge we now have the `DatadogAgent.Spec.Features` section, I think it cool be good to move this configuration in the `features` section

Reply:
Also about the name, I'm wondering if we can name it: `KubeStateMetrics`

Reply:
We have been introducing the new check as a core check, I think it is important to distinguish that it is not the V1.
I tried changing out of curiosity, and it gets confusing with what is KSM, the conf of the check and the check runtime itself.

Reply:
I split up in 2 commits, the last one refactors the code to use the feature. Let me know which one you prefer (can revert the last commit if necessary, I actually like using the `features`)

Reply:
The doc is great 👍 , just few nits.

For the name, `KubeStateMetrics` or `KubeStateMetricsCore` is not a big deal. I just wanted to have a same that talk to the user in term of a feature. 

For the configuration in the CRD, I added another comment, let me know what do you think.